### ISSUE TITLE
Support SASL authentication to Kafka

### ISSUE THREAD

**Issue by coltmcnealy-lh** [littlehorse-enterprises/littlehorse#713]
Currently, we only support MTLS authentication to Kafka. We should support `SASL_PLAIN`, since this is what is used by Confluent Cloud.

## New Configurations

It is only in scope to support `SASL_PLAIN` for this ticket. Therefore, we will add the following configurations:

* `LHS_KAFKA_SASL_PLAIN_USERNAME`: The SASL username
* `LHS_KAFKA_SASL_PLAIN_PASSWORD`: The SASL Password
* `LHS_KAFKA_SASL_PLAIN_PASSWORD_FILE`: The file in which the password is stored.

## Considerations

* It should be possible to use `SASL_PLAIN` along with TLS.

**Comment by coltmcnealy-lh** [comment]
It appears that we also need to introduce another configuration: `LHS_KAFKA_SECURITY_PROTOCOL`.

Why? Well, we currently don't support the SSL protocol option (i.e. not `PLAINTEXT` and not MTLS).

This configuration can't actually be inferred from the other configurations, because there is no way to determine between `SSL` and `PLAINTEXT` in the case that the Kafka brokers are using a publicly trusted certificate (in this case, we wouldn't need to set `LHS_KAFKA_TRUSTSTORE`).

There is also a bug in the current state: if the user _does_ set the `LHS_KAFKA_TRUSTSTORE` and does not also set the Keystore (used for MTLS), then an `LHMisconfigurationException` is thrown. 

The TLDR:
* Currently, we don't support `SSL` listeners on Kafka (and we should)
* We need to introduce a new configuration `LHS_KAFKA_SECURITY_PROTOCOL` which defaults to `PLAINTEXT` and is passed through to the Kafka native `security.protocol` configuration.

**Comment by coltmcnealy-lh** [comment]
Further investigation yields that the `sasl.jaas.config` configuration is a bit more nuanced than I orignially thought, and could vary drasitcally depending on the exact use-case. For Strimzi, it looks like:

* `org.apache.kafka.common.security.scram.ScramLoginModule required username="basic-tls-lh-kafka" password=xxxxxxxx`

For Confluent, it is:
* `org.apache.kafka.common.security.scram.PlainLoginModule required username="basic-tls-lh-kafka" password=xxxxxxxx`

I propose now changing the implementation to have:
* `LHS_KAFKA_SASL_JAAS_CONFIG`
* `LHS_KAFKA_SASL_MECHANISM`

We should allow the user the flexibility to set those configurations properly.

**Comment by coltmcnealy-lh** [comment]
Finished by #716
